test/puppetlabs/puppetdb/command_test.clj

```diff
@@ -957,17 +993,19 @@
                                :version 4
                                :payload new-facts}]
 
-            (test-msg-handler new-facts-cmd publish discard-dir
-              (reset! second-message? true)
-              (let [[call :as calls] (args-supplied publish)
-                    [msg] call
-                    attempts (:attempts msg)]
-                (is (= 1 (count calls)))
-                (is (= 1 (count call)))
-                (is (= 1 (count attempts)))
-                (is (re-matches
-                     #"(?sm).*ERROR: could not serialize access due to concurrent update.*"
-                     (-> attempts first :exception full-sql-exception-msg)))))
+            (handle-message (store-command' q new-facts-cmd))
+            (reset! second-message? true)
+            (let [failed-cmdref (async/<!! command-chan)]
```

Comment:
So as discussed, potentially having a timeout here will alleviate any blocking issues in tests, even if it is stupidly big. A stack trace early, will show us the line of code thats failing, but a permanent blocking call, will not show us a trace, so solving a lost command scenario will be harder. Its probably not going to happen, just thinking about how one can defend from this kind of thing.